### ISSUE TITLE
token_type typo in validate.py audit

### ISSUE THREAD

**Issue by quynhnhatnguyen** [privacyidea/privacyidea#1846]
Hi @cornelinux,

It seems that there is a typo in the validate.py for `token_type` param, where it is spelled as `tokentype` instead.

For example:
```
    g.audit_object.log({"info": log_used_user(user, details.get("message")),
                        "success": result,
                        "serial": serial or details.get("serial"),
                        "tokentype": details.get("type")})
```

Correct one should be:
```
    g.audit_object.log({"info": log_used_user(user, details.get("message")),
                        "success": result,
                        "serial": serial or details.get("serial"),
                        "token_type": details.get("type")})
```

This cause the token type is not in our audit. There are multiple occurrence of this typos not just the above block of code.

Can we please confirm?

Best regards,

Quynh Nguyen

**Comment by cornelinux** [comment]
Hello @quynhnhatnguyen 
you are right. The correct key should be **token_type** with an underscore.
I can see that there is the wrong key in ``/validate/check`` and ``/validate/samlcheck``.
and in ``api/system.py::test``

If you see any further occurrences, please add them!

Thanks a lot for pointing this out. We will fix this in version 3.1.1.

**Comment by fredreichbier** [comment]
This was fixed by dd6ae2f in branch-3.1, but hasn't been merged in master yet. Thanks for the report!

**Comment by fredreichbier** [comment]
#1855 merged this into master.
